File: Spring/les/src/main/java/com/jga/les/model/Permissao.java
package com.jga.les.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Permissao{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Tela tela;

    private boolean add;
    private boolean edit;
    private boolean delete;

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;

        Permissao permissao = (Permissao) o;
        return tela.equals(permissao.tela);
    }
}

File: Flutter/les/lib/model/usuario/permissao.dart

import 'package:les/model/entity.dart';
import 'package:les/model/usuario/tela.dart';

class Permissao extends Entity{

  final Tela tela;
  bool add;
  bool edit;
  bool delete;

  Permissao({super.id, required this.tela, required this.add, required this.edit, required this.delete});

  factory Permissao.fromJson(Map<String, dynamic> json){
    return Permissao(
        id: json['id'],
        tela: Tela.fromJson(json['tela']),
        add: json['add'],
        edit: json['edit'],
        delete: json['delete']
    );
  }

  @override
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tela': tela.toJson(),
      'add': add,
      'edit': edit,
      'delete': delete
    };
  }
}